streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -452,9 +463,14 @@ private void runLoop() {
                         consumer.seekToEnd(ex.partitions());
                         log.info(String.format("stream-thread [%s] setting topic to consume from latest offset %s", this.getName(), partition.topic()));
                     } else {
-                        //If not defined go to default config of earliest
-                        consumer.seekToBeginning(ex.partitions());
-                        log.info(String.format("stream-thread [%s] no setting defined for topic %s setting to consume from earliest offset", this.getName(), partition.topic()));
+                        if (originalReset != null) { //client could specify "none" and not provide custom offset resets
+                            if (originalReset.equals("earliest")) {
+                                consumer.seekToBeginning(ex.partitions());
+                            } else if (originalReset.equals("latest")) {
+                                consumer.seekToEnd(ex.partitions());
+                            }
+                            log.info(String.format("stream-thread [%s] no custom setting defined for topic %s using original config  %s for offset reset", this.getName(), partition.topic(), originalReset));
```

Comment:
We also need to throw here. See comment below. Please unify both `throw`s to avoid code duplication.